pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,178 @@
+        else:
+            if wires is None:
+                raise ValueError("Please ensure wires is specified.")
+            if wires is not None and set(wires) != set(graph.nodes):
+                raise ValueError("Please ensure wires objects match labels in graph")
+            super().__init__(wires=wires)
+
+    def label(self) -> str:  # pylint: disable=arguments-differ
+        r"""Defines how the graph state preparation is represented in diagrams and drawings.
+
+        Returns:
+            str: label to use in drawings
+        """
+        return self.__repr__()
+
+    def __repr__(self):
+        """Method defining the string representation of this class."""
+        return f"GraphStatePrep({self.hyperparameters['one_qubit_ops'](wires=0).name}, {self.hyperparameters['two_qubit_ops'].name})"
+
+    def decomposition(self) -> list["Operator"]:
+        r"""Representation of the operator as a product of other operators.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+        return self.compute_decomposition(wires=self.wires, **self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(
+        wires: Wires,
+        graph: Union[nx.Graph, QubitGraph],
+        one_qubit_ops: Operation = qml.H,
+        two_qubit_ops: Operation = qml.CZ,
+    ):  # pylint: disable=arguments-differ, unused-argument
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. note::
+
+            Operations making up the decomposition should be queued within the
+            ``compute_decomposition`` method.
+
+        .. seealso:: :meth:`~.Operator.decomposition`.
+
+        Args:
+            wires (Wires): Wires the decomposition applies on.
+            graph (Union[nx.Graph, QubitGraph]): QubitGraph or nx.Graph object mapping qubit to wires.
+            one_qubit_ops (Operation): Operator to prepare the initial state of each qubit. Default to :class:`~.pennylane.H`.
+            two_qubit_ops (Operation): Operator to entangle nearest qubits. Default to :class:`~.pennylane.CZ`.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+
+        op_list = []
+
+        # Add two_qubit_ops for each pair of nearest qubits in the graph
+        if isinstance(graph, QubitGraph):
+            # Add one_qubit_ops for each qubit in the graph
+            for wire in wires:
+                op_list.append(one_qubit_ops(wires=graph[wire]))
+
+            for qubit0, qubit1 in graph.graph.edges:
+                op_list.append(two_qubit_ops(wires=[graph[qubit0], graph[qubit1]]))
+        else:
+            # Add one_qubit_ops for each qubit in the graph
+            for wire in wires:
+                op_list.append(one_qubit_ops(wires=wire))
+
+            for qubit0, qubit1 in graph.edges:
+                op_list.append(two_qubit_ops(wires=[qubit0, qubit1]))
```

Comment:
```suggestion
        edges = graph.edge_labels if isinstance(graph, QubitGraph) else graph.edges
        nodes = graph.node_labels if isinstance(graph, QubitGraph) else graph.nodes

        if set(wires) != set(nodes):
            wire_map = dict(zip(nodes, wires))
            edges = [(wire_map[edge[0]], wire_map[edge[1]]) for edge in edges]

        for wire in wires:
            op_list.append(one_qubit_ops(wires=wire))
        for edge in edges:
            op_list.append(two_qubit_ops(wires=edge))
```

Mapping like this should allow us to pass string and integer wires that don't match the node labels. In the static `compute_decomposition` method, we assume wires is not None, so this should work for a graph:

```
lattice = generate_lattice([2, 2], "square")
qgraph = QubitGraph(lattice.graph, "square")
```

inputs that should work for `(wires, graph)`:

```
(set(qubit_graph.children), qubit_graph)
([1, 2, 'c', 4], qubit_graph)
([1, 2, 'c', 4], qubit_graph.graph)
```